tools/freeze-requirements.sh

```diff
@@ -40,6 +40,10 @@ pip install -e .
 pip freeze --exclude-editable > requirements.txt
 requirements_fixups "requirements.txt"
 
+pip install -e .[docs]
```

Comment:
I know this is how we do it in charmcraft and rockcraft, but I'm not exactly thrilled with having the docs requirements added to the dev requirements in general.

Reply:
What's the potential negative impact?

Reply:
If we have tests or something that use the docs requirements, it'll be hidden that they need to be in the dev requirements too. It also means that `tox` will take longer, since it's having to install each of those requirements in each test environment.